Marshal/JSON.swift

```diff
@@ -60,3 +60,10 @@ extension Dictionary : JSONCollectionType {}
 extension Array : JSONCollectionType {}
 
 extension Set : JSONCollectionType {}
+
+
+// MARK: - Marshaling
+
+public protocol JSONMarshaling {
```

Comment:
@bwhiteley it didn't like the method name to be the same case as the data type, so this is what I made instead.